Dockerfile

```diff
@@ -1,7 +1,40 @@
-FROM ubuntu:latest
-WORKDIR /app
-COPY ./target/release/prover-server .
-COPY ./libzktrie.so .
-ENV LD_LIBRARY_PATH=/app/libzktrie.so:
+FROM zktachyon/halo2:jammy-latest AS builder
+LABEL maintainer="Kroma <kroma@kroma.network>"
+
+ARG RUST_TOOLCHAIN_VERSION=nightly-2022-12-10
+
+RUN apt update && \
+    apt install -y --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    curl \
+    libclang-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain ${RUST_TOOLCHAIN_VERSION} -y
+ENV PATH="/root/.cargo/bin:${PATH}"
+
+RUN curl -LO "https://golang.org/dl/go1.20.3.linux-amd64.tar.gz" && \
+    rm -rf /usr/local/go && tar -C /usr/local -xzf go1.20.3.linux-amd64.tar.gz
+ENV PATH="${PATH}:/usr/local/go/bin"
+
+COPY . /usr/src/kroma-prover
+WORKDIR /usr/src/kroma-prover
+
+RUN cargo build --release --bin prover-server --features=tachyon
+RUN mkdir -p /temp-lib && \
+    cp $(find /usr/lib/ -name libtachyon.so.0) /temp-lib/ && \
+    cp $(find ./target/release/ -name libzktrie.so) /temp-lib/ && \
+    cp $(find /usr/lib/ -name libgomp.so.1) /temp-lib/ && \
+    cp $(find /usr/lib/ -name libgmpxx.so.4) /temp-lib/
+
+FROM ubuntu:jammy AS kroma-prover
+LABEL maintainer="Kroma <kroma@kroma.network>"
+
```

Comment:
```suggestion

RUN apt update && \
    apt install -y --no-install-recommends \
    libgmp10 \
    libomp5 && \
    rm -rf /var/lib/apt/lists/*
    
```

Reply:
I've installed libgmp-dev since I was getting
```
./prover-server: error while loading shared libraries: libgmpxx.so.4: cannot open shared object file: No such file or directory
```
error, and libgmp-dev package has libgmp10 and libgmpxx4ldbl at the same time!

Also I install libgomp1 instead since I was getting

```
./prover-server: error while loading shared libraries: libgomp.so.1: cannot open shared object file: No such file or directory
```
error!